runelite-client/src/main/java/net/runelite/client/plugins/config/PluginListItem.java

```diff
@@ -0,0 +1,300 @@
+				CONFIG_ICON = new ImageIcon(configIcon);
+				CONFIG_ICON_HOVER = new ImageIcon(SwingUtil.grayscaleOffset(configIcon, -100));
+				ON_SWITCHER = new ImageIcon(ImageIO.read(ConfigPanel.class.getResourceAsStream("switchers/on.png")));
+				OFF_SWITCHER = new ImageIcon(ImageIO.read(ConfigPanel.class.getResourceAsStream("switchers/off.png")));
+				ON_STAR = new ImageIcon(ImageIO.read(ConfigPanel.class.getResourceAsStream("stars/on.png")));
+				OFF_STAR = new ImageIcon(ImageIO.read(ConfigPanel.class.getResourceAsStream("stars/off.png")));
+			}
+		}
+		catch (IOException e)
+		{
+			throw new RuntimeException(e);
+		}
+	}
+
+	/**
+	 * Creates a new {@code PluginListItem} for a plugin.
+	 * <p>
+	 * Note that {@code config} and {@code configDescriptor} can be {@code null}
+	 * if there is no configuration associated with the plugin.
+	 */
+	PluginListItem(ConfigPanel configPanel, Plugin plugin, PluginDescriptor descriptor,
+		@Nullable Config config, @Nullable ConfigDescriptor configDescriptor)
+	{
+		this(configPanel, plugin, config, configDescriptor,
+			descriptor.name(), descriptor.description(), descriptor.tags());
+	}
+
+	/**
+	 * Creates a new {@code PluginListItem} for a core configuration.
+	 */
+	PluginListItem(ConfigPanel configPanel, Config config, ConfigDescriptor configDescriptor,
+		String name, String description, String... tags)
+	{
+		this(configPanel, null, config, configDescriptor, name, description, tags);
+	}
+
+	private PluginListItem(ConfigPanel configPanel, @Nullable Plugin plugin, @Nullable Config config,
+		@Nullable ConfigDescriptor configDescriptor, String name, String description, String... tags)
+	{
+		this.configPanel = configPanel;
+		this.plugin = plugin;
+		this.config = config;
+		this.configDescriptor = configDescriptor;
+		this.name = name;
+		this.description = description;
+		Collections.addAll(keywords, name.toLowerCase().split(" "));
+		Collections.addAll(keywords, tags);
+
+		setLayout(new BorderLayout(3, 0));
+		setPreferredSize(new Dimension(PluginPanel.PANEL_WIDTH, 20));
+
+		JLabel nameLabel = new JLabel(name);
+		nameLabel.setForeground(Color.WHITE);
+
+		if (!description.isEmpty())
+		{
+			nameLabel.setToolTipText("<html>" + name + ":<br>" + description + "</html>");
+		}
+
+		add(nameLabel, BorderLayout.CENTER);
+
+		pinButton.setPreferredSize(new Dimension(21, 0));
+		attachPinButtonListener();
+		add(pinButton, BorderLayout.LINE_START);
+
+		final JPanel buttonPanel = new JPanel();
+		buttonPanel.setLayout(new GridLayout(1, 2));
+		add(buttonPanel, BorderLayout.LINE_END);
+
+		configButton.setPreferredSize(new Dimension(25, 0));
+		configButton.setVisible(false);
+		attachConfigButtonListener();
+		buttonPanel.add(configButton);
+
+		toggleButton.setPreferredSize(new Dimension(25, 0));
+		attachToggleButtonListener(toggleButton);
+		buttonPanel.add(toggleButton);
+	}
+
+	private void attachPinButtonListener()
+	{
+		pinButton.addMouseListener(new MouseAdapter()
```

Comment:
Prefer ActionListener to MouseAdapter as it works with keyboards. If you must use a MouseAdapter use mouseClicked rather than mousePressed